src/chmatch.c

```diff
@@ -52,19 +44,17 @@ static SEXP chmatchMain(SEXP x, SEXP table, int nomatch, bool chin, bool chmatch
     SET_TRUELENGTH(s,0);   // TODO: do we need to set to zero first (we can rely on R 3.1.0 now)?
   }
   const int tablelen = length(table);
-  const SEXP *td = STRING_PTR(table);
```

Comment:
This allowed to reduce number of STRING_ELT calls below. Cannot we have that back using on utable?

Reply:
I've made the change. Now the git diff lines are reduced as well. Thanks.